iOS_SDK/OneSignalSDK/Source/OneSignalLiveActivityController.m

```diff
@@ -0,0 +1,165 @@
+//
+//  OneSignalLiveActivityController.m
+//  OneSignalCore
+//
+//  Created by Henry Boswell on 1/11/23.
+//  Copyright © 2023 Hiptic. All rights reserved.
+//
+
+#import <Foundation/Foundation.h>
+#import <OneSignalCore/OneSignalCore.h>
+
+#import "OneSignalLiveActivityController.h"
+
+@interface OSPendingLiveActivityUpdate: NSObject
+    @property NSString* activityId;
+    @property NSString* appId;
+    @property NSString* token;
+    @property BOOL isEnter;
+    @property OSResultSuccessBlock successBlock;
+    @property OSFailureBlock failureBlock;
+    - (id)initWith:(NSString * _Nonnull)activityId
+         appId:(NSString * _Nonnull)appId
+         withToken:(NSString * _Nonnull)token
+           isEnter:(BOOL)isEnter
+       withSuccess:(OSResultSuccessBlock _Nullable)successBlock
+       withFailure:(OSFailureBlock _Nullable)failureBlock;
+@end
+
+@implementation OSPendingLiveActivityUpdate
+
+- (id)initWith:(NSString * _Nonnull)activityId
+         appId:(NSString * _Nonnull)appId
+     withToken:(NSString *)token
+       isEnter:(BOOL)isEnter
+   withSuccess:(OSResultSuccessBlock)successBlock
+   withFailure:(OSFailureBlock)failureBlock {
+    self.token = token;
+    self.activityId = activityId;
+    self.appId = appId;
+    self.isEnter = isEnter;
+    self.successBlock = successBlock;
+    self.failureBlock = failureBlock;
+    return self;
+};
+@end
+
+@implementation OneSignalLiveActivityController
+
+static NSMutableArray* pendingLiveActivityUpdates;
+static NSString* subscriptionId;
+
++ (void) initialize {
+    subscriptionId = OneSignalUserManagerImpl.sharedInstance.pushSubscriptionId;
```

Comment:
You'll need to add `OneSignalLiveActivityController` as an observer of the push subscription with `[OneSignalUserManagerImpl.sharedInstance addObserver:self]` or some singleton of `OneSignalLiveActivityController` or else `onOSPushSubscriptionChangedWithStateChanges` will never fire for this class.

Reply:
Done